src/main.c

```diff
@@ -223,15 +240,219 @@ static int init_context(int argc, char **argv, const struct zl_config_t *cfg) {
+    get_yaml_pair_key(doc, pair, key, sizeof(key));
+    if (0 == strcmp(key, name)) {
+      value_node = yaml_document_get_node(doc, pair->value);
+      break;
+    }
+  }
+  return value_node;
+}
+
+static void get_int_array_from_yaml_sequence(yaml_document_t *doc, yaml_node_t *node, zl_int_array_t *data) {
+  char buf[ZL_YAML_KEY_LEN + 1];
+  for (yaml_node_item_t *item = node->data.sequence.items.start; item != node->data.sequence.items.top; item++) {
+    get_yaml_item(doc, *item, buf, sizeof(buf));
+    if (data->count < ZL_INT_ARRAY_CAPACITY) {
+      data->data[data->count++] = atoi(buf);
+    } else {
+      WARN ("yaml sequence is too large\n");
+      break;
+    }
+  }
+}
+
+static yaml_node_t *get_node_by_path(yaml_document_t *doc, yaml_node_t *node, const char **path, size_t path_len) {
+  for (size_t i = 0; i < path_len; i++) {
+    node = get_child_node(doc, node, path[i]);
+    if (!node) {
+      break;
+    }
+  }
+  return node;
+}
+
+static int get_int_by_path(yaml_document_t *doc, yaml_node_t *root, const char **path, size_t path_len, int *result) {
+  yaml_node_t *node = get_node_by_path(doc, root, path, path_len);
+  if (node && node->type == YAML_SCALAR_NODE) {
+    char buf[128];
+    get_yaml_scalar(doc, node, buf, sizeof(buf));
+    *result = atoi(buf);
+    return 0;
+  }
+  return -1;
+}
+
+static int get_string_by_yaml_path(yaml_document_t *doc, yaml_node_t *root, const char **path, size_t path_len, char *buf, int buf_size) {
+  yaml_node_t *node = get_node_by_path(doc, root, path, path_len);
+  if (node && node->type == YAML_SCALAR_NODE) {
+    get_yaml_scalar(doc, node, buf, buf_size);
+    return 0;
+  }
+  return -1;
+}
+
+static int get_int_array_by_path(yaml_document_t *doc, yaml_node_t *root, const char **path, size_t path_len, zl_int_array_t *arr) {
+  yaml_node_t *node = get_node_by_path(doc, root, path, path_len);
+  if (node && node->type == YAML_SEQUENCE_NODE) {
+    get_int_array_from_yaml_sequence(doc, node, arr);
+    return 0;
+  }
+  return -1;
+}
+
+static void snprint_int_array(zl_int_array_t *array, char *buf, size_t buf_size) {
+  int pos = 0;
+  for (int i = 0; i < array->count; i++) {
+    pos += snprintf(buf + pos, buf_size - pos, "%d%s", array->data[i], i == array->count - 1 ? "" : " ");
+  }
+}
+
+static void init_component_restart_intervals(zl_comp_t *comp) {
+  zl_yaml_config_t *zowe_yaml_config = &zl_context.yaml_config;
+  yaml_document_t *document = &zowe_yaml_config->document;
+  DEBUG ("loading restart intervals for component '%s'\n", comp->name);
+  yaml_node_t *root = zowe_yaml_config->root;
+  const char *zowe_path[] = {"zowe", "launcher", "restartIntervals"};
+  const char *component_path[] = {"components", comp->name, "launcher", "restartIntervals"};
+  const char *instance_path[] = {"haInstances", zl_context.ha_instance_id, "components", comp->name, "launcher", "restartIntervals"};
```

Comment:
Hi Leonty, is that possible to change `zl_context.ha_instance_id` to lower case? Currently in my test, it's not picking up the haInstances customization.

JES always converts this value to upper case, but in `zowe.yaml` I actually use lower case to define. It was a dilemma for me if should use upper case as HA INSTANCE ID in zowe.yaml, but I chose to use lower case since it looks nicer. I should document this part better. Or make it case insensitive everywhere.

Please let me know what's your thoughts. Thanks

Reply:
@jackjia-ibm sure, let's convert `ha_instance_id` to lowercase. In the future, we can opt for insensitive comparison if we really need it.